code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -37,41 +39,45 @@ There are several things that need to be remembered:
 	slime etc. Instead, it'll just return without doing any work. So no harm in calling it for slimes and such.
 
 
-> There are also these special cases:
-		update_damage_overlays() //handles damage overlays for brute/burn damage
-		update_body() //Handles updating your mob's body layer and mutant bodyparts
+>	There are also these special cases:
+		update_damage_overlays()	//handles damage overlays for brute/burn damage
+		update_body()				//Handles updating your mob's body layer and mutant bodyparts
 									as well as sprite-accessories that didn't really fit elsewhere (underwear, undershirts, socks, lips, eyes)
 									//NOTE: update_mutantrace() is now merged into this!
-		update_hair() //Handles updating your hair overlay (used to be update_face, but mouth and
+		update_hair()				//Handles updating your hair overlay (used to be update_face, but mouth and
 									eyes were merged into update_body())
 
 
 */
 
 //HAIR OVERLAY
-/mob/living/carbon/human/update_hair()
-	dna.species.handle_hair(src)
+/mob/living/carbon/human/update_hair(is_creating)
+	var/obj/item/bodypart/head/my_head = get_bodypart(BODY_ZONE_HEAD)
+	if(!my_head)
+		return
+	my_head.update_limb(FALSE, is_creating)
+	update_body_parts()
```

Comment:
why do you need to update all body parts when only hair/the head is being updated?

Reply:
update_limb doesnt redraw the limb, it sets its infomation. update_body_parts is the only way to actually draw the body.